### PR TITLE
[WIP] Including the api providers in webb.js and adding unit/integration tests

### PR DESCRIPTION
**Pull request by AhmedKorim**
# overview
Moving the providers from the dApp `react-environment` package to webb.js for adding more unit/integrations tests, and making the dApp leaner

### CODE REVIEW

File: packages/api-providers/src/contracts/abis/BadRecipient.json [hunk 1]
```diff
@@ -0,0 +1,372 @@
+{
```
Comment (drewstone) [inline packages/api-providers/src/contracts/abis/BadRecipient.json:1]: Is this file needed?

File: packages/api-providers/src/webb-context/bridge/bridge-deposit.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { DepositPayload, MixerDeposit } from '../mixer/mixer-deposit';
+import { WebbApiProvider } from '../webb-provider.interface';
+
+export abstract class BridgeDeposit<
```
Comment (drewstone) [inline packages/api-providers/src/webb-context/bridge/bridge-deposit.ts:4]: Can we consider using `AnchorDeposit`? And changing the directory to `anchor/anchor-deposit`

File: packages/api-providers/src/webb-context/bridge/bridge-deposit.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  T extends WebbApiProvider<any>,
+  K extends DepositPayload = DepositPayload<any>
+> extends MixerDeposit<T, K> {
+  generateNote(mixerId: number | string): Promise<K> {
+    throw new Error('api not ready:Not mixer api');
+  }
+
+  abstract generateBridgeNote(
```
Comment (drewstone) [inline packages/api-providers/src/webb-context/bridge/bridge-deposit.ts:12]: `generateAnchorNote`

File: packages/api-providers/src/webb-context/bridge/bridge-deposit.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    mixerId: number | string,
```
Comment (drewstone) [inline packages/api-providers/src/webb-context/bridge/bridge-deposit.ts:13]: `anchorId`

File: packages/api-providers/src/webb-context/bridge/bridge-deposit.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { DepositPayload, MixerDeposit } from '../mixer/mixer-deposit';
+import { WebbApiProvider } from '../webb-provider.interface';
+
+export abstract class BridgeDeposit<
+  T extends WebbApiProvider<any>,
+  K extends DepositPayload = DepositPayload<any>
+> extends MixerDeposit<T, K> {
+  generateNote(mixerId: number | string): Promise<K> {
```
Comment (drewstone) [inline packages/api-providers/src/webb-context/bridge/bridge-deposit.ts:8]: `anchorId`

File: packages/api-providers/src/webb-context/bridge/bridge.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import { Currency } from '../currency/currency';
+import { BridgeConfig } from '../../types/bridge-config.interface';
+import { AppConfig } from '../common';
+import { WebbCurrencyId } from '../../enums';
+import { InternalChainId } from '../../chains';
+import { CurrencyRole } from '../../types/currency-config.interface';
+
+export class Bridge {
```
Comment (drewstone) [inline packages/api-providers/src/webb-context/bridge/bridge.ts:8]: This is OK, the directory should be changed to `anchor/bridge.ts` because ultimately this is a bridge between anchors.

File: packages/api-providers/src/bridge/bridge-withdraw.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import { Bridge, MixerWithdraw, WebbApiProvider } from '../webb-context';
+import { InternalChainId } from '../chains';
+
+export abstract class BridgeWithdraw<T extends WebbApiProvider<any>> extends MixerWithdraw<T> {
```
Comment (drewstone) [inline packages/api-providers/src/bridge/bridge-withdraw.ts:4]: This should be `AnchorWithdraw`

File: packages/api-providers/src/contracts/abis/BadRecipient.json (hunk 1 above)
Comment (AhmedKorim) [reply packages/api-providers/src/contracts/abis/BadRecipient.json:1]: Will sync with @nathan on the integration of `@webb-tools/contracts`

File: packages/api-providers/src/contracts/circuits/withdraw_verification_key.json [hunk 8]
```diff
@@ -0,0 +1 @@
...87432"],["11434086686358152335540554643130007307617078324975981257823476472104616196090","1081836006956609894549771334721413187913047383331561601606260283167615953295"],["1","0"]],"vk_alfabeta_12":[[["21365812195485565970449951947073093780266368302919758441364015846541070649889","13899346400535634506212110533298291334607392402427556671205120154633328392938"],["9176241749501911223800029754421468853554325976185319022570714890557573371905","16897971416518120303319064173781767941602184168568029998374699041158871223961"],["6994180461635063460601079479427022670717591990958754250719881638298446317234","9179026066977875182617507817914233459615244337807200435373777462238862437279"]],[["12505271218650914527710842094803759916526791354425688234288687823807495160149","21079529389819175768418925187724856376731616978516023985093353832725307191172"],["17191503117818720549103434672697266076462686591305810428253469897937793366640","1460614254014044318793914579722356842222906714283817829623699739748310555420"],["8156206509351086444845867552906130156524383450240899694386232124211773153798","9170252275562998399982022993666567730500187520875477300499051353566398457339"]]],"protocol":"groth","nPublic":6}
```
[the remark this answers is not in the document]
Comment (AhmedKorim) [reply packages/api-providers/src/contracts/circuits/withdraw_verification_key.json:1]: @nepoche  cc

File: packages/api-providers/src/chains/chain-utils.spec.ts [hunk 9]
```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 @webb-tools/
+// SPDX-License-Identifier: Apache-2.0
+
+import { expect } from 'chai';
+
+import { ChainType, ChainTypeId } from './chain-id.enum';
+import { byteArrayToNum, computeChainIdType, numToByteArray, parseChainIdType } from './chain-utils';
+
+// TODO: Move utils and tests to webb.js
```
Comment (nepoche) [inline packages/api-providers/src/chains/chain-utils.spec.ts:9]:
```suggestion
```

File: packages/sdk-core/src/proving/proving-manager.ts [hunk 10]
```diff
@@ -1,23 +1,39 @@
...
+    const worker = this.worker;
+
+    if (worker) {
+      return ProvingManager.proveWithWorker(input, worker);
+    }
+
+    return ProvingManager.proofWithoutWorker(input);
+  }
+
+  private static proofWithoutWorker (input: ProvingManagerSetupInput) {
```
Comment (nepoche) [inline packages/sdk-core/src/proving/proving-manager.ts:22]:
```suggestion
  private static proveWithoutWorker (input: ProvingManagerSetupInput) {
```

File: packages/api-providers/src/contracts/wrappers/anchor-interface.ts [hunk 11]
```diff
@@ -0,0 +1,9 @@
+// Copyright 2022 @webb-tools/
```
[the remark this answers is not in the document]
Comment (AhmedKorim) [reply packages/api-providers/src/contracts/wrappers/anchor-interface.ts:1]: This is still used for Tornado anchors, let' see what we can do about it on another PR

File: packages/api-providers/src/contracts/utils/evm-note.ts [hunk 12]
```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 @webb-tools/
```
[the remark this answers is not in the document]
Comment (AhmedKorim) [reply packages/api-providers/src/contracts/utils/evm-note.ts:1]: This is still used for Tornado anchors, let' see what we can do about it on another PR
